tools/evaluate-sroie.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python
+
+from argparse import ArgumentParser
+from pathlib import Path
+from subprocess import run
+import datasets
+import time
+import tempfile
+from tqdm import tqdm
+
+from sklearn.feature_extraction.text import CountVectorizer
+from sklearn.metrics import f1_score, precision_score, recall_score
+
+try:
+    import pytesseract
+except ImportError:
+    pytesseract = None
+
+
+def build_ocrs() -> None:
+    run("cargo build --release -p ocrs-cli", shell=True, check=True, text=True)
+
+
+def extract_text(image_path: str) -> str:
+    """Extract text from an image using ocrs."""
+    result = run(
+        # We run the binary directly here rather than use `cargo run` as it
+        # is slightly faster.
+        [f"target/release/ocrs", image_path],
+        check=True,
+        text=True,
+        capture_output=True,
+    )
+    return result.stdout
+
+
+def run_global_retrieval_eval(max_samples: int) -> bool:
+    """
+    Evaluate OCR performance, by computing precision, recall and F1 score
+    for the detected tokens globally on the whole document
+
+    Here we use scikit-learn's tokenizer to split the text into tokens
+    """
+
+    # Evaluate the SROIE dataset
+    dataset = datasets.load_dataset("rth/sroie-2019-v2", split="test")
+    true_text = ["\n".join(el["objects"]["text"]) for el in dataset]
+    print("Evaluating on SROIE 2019 dataset...")
+
+    # Build the vocabulary on the ground truth
+    vectorizer = CountVectorizer(input="content", binary=True)
+
+    X_true = vectorizer.fit_transform(true_text[:max_samples])
+
+    # Evaluate with orcs
+    text_pred_orcs = []
+    time_orcs = 0
+
+    # Use a tempfs if available (Linux, MacOS) to reduce disk I/O overhead
+    TMP_DIR = Path("/dev/shm")
+    if not TMP_DIR.exists():
+        TMP_DIR = None
```

Comment:
You, are right it doesn't work on Mac. It's a few milisecond or less depending on SSD performance, it was more to avoid unnecessairly write stuff to disk. Will remove.